internal/client/client.go

```diff
@@ -175,126 +175,128 @@ func (c *CSAPI) SyncUntil(t *testing.T, since, key string, check func(gjson.Resu
 	}
 }
 
-// MustDoWithStatus is the same as MustDo but fails the test if the response code does not match that provided
-func (c *CSAPI) MustDoWithStatus(t *testing.T, method string, paths []string, jsonBody interface{}, status int) *http.Response {
-	t.Helper()
-	res, err := c.DoWithAuth(t, method, paths, jsonBody)
-	if err != nil {
-		t.Fatalf("CSAPI.MustDoWithStatus %s %s error: %s", method, strings.Join(paths, "/"), err)
-	}
-	if res.StatusCode != status {
-		t.Fatalf("CSAPI.MustDoWithStatus %s %s returned HTTP %d, expected %d", method, res.Request.URL.String(), res.StatusCode, status)
-	}
-	return res
-}
-
-// MustDoWithStatusRaw is the same as MustDoRaw but fails the test if the response code does not match that provided
-func (c *CSAPI) MustDoWithStatusRaw(t *testing.T, method string, paths []string, body []byte, contentType string, query url.Values, status int) *http.Response {
-	t.Helper()
-	res, err := c.DoWithAuthRaw(t, method, paths, body, contentType, query)
-	if err != nil {
-		t.Fatalf("CSAPI.MustDoWithStatusRaw %s %s error: %s", method, strings.Join(paths, "/"), err)
-	}
-	if res.StatusCode != status {
-		t.Fatalf("CSAPI.MustDoWithStatusRaw %s %s returned HTTP %d, expected %d", method, res.Request.URL.String(), res.StatusCode, status)
-	}
-	return res
-}
-
-// MustDo is the same as Do but fails the test if the response is not 2xx
+// MustDo will do the HTTP request and fail the test if the response is not 2xx
 func (c *CSAPI) MustDo(t *testing.T, method string, paths []string, jsonBody interface{}) *http.Response {
```

Comment:
Is there a point to keeping `MustDo` around while `MustDoFunc` exists? I noticed that we didn't keep `Do` around while `DoFunc` exists.

Reply:
Yeah because I don't want to convert them all in one PR. It makes the diff too big

Reply:
Ah alright, good to know the plan is to axe it eventually.